src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest/PublishVerticalManifest.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest
+{
+    public class PublishVerticalManifest : Task
+    {
+        [Required]
+        public ITaskItem[] AssetManifest { get; set; }
+        [Required]
+        public string VerticalAssetManifestOutputPath { get; set; }
+        [Required]
+        public string VmrBuildNumber { get; set; }
+
+        private static readonly string _buildIdAttribute = "BuildId";
+        private static readonly string _azureDevOpsBuildNumberAttribute = "AzureDevOpsBuildNumber";
+        private static readonly string[] _ignoredAttributes = new string[] { _buildIdAttribute, _azureDevOpsBuildNumberAttribute, "IsReleaseOnlyPackageVersion" };
+
+        public override bool Execute()
+        {
+            var assetManifestXmls = AssetManifest.Select(xmlPath => XDocument.Load(xmlPath.ItemSpec)).ToList();
+
+            VerifyAssetManifests(assetManifestXmls);
+
+            var verticalManifestRoot = assetManifestXmls.First().Root;
+
+            // Set the BuildId and AzureDevOpsBuildNumber attributes to the value of VmrBuildNumber
+            verticalManifestRoot.SetAttributeValue(_buildIdAttribute, VmrBuildNumber);
+            verticalManifestRoot.SetAttributeValue(_azureDevOpsBuildNumberAttribute, VmrBuildNumber);
+
+            var packageElements = new List<XElement>();
+            var blobElements = new List<XElement>();
+
+            foreach (var assetManifestXml in assetManifestXmls)
+            {
+                packageElements.AddRange(assetManifestXml.Descendants("Package"));
+                blobElements.AddRange(assetManifestXml.Descendants("Blob"));
+            }
+            
+            packageElements = packageElements.OrderBy(packageElement => packageElement.Attribute("Id").Value).ToList();
+            blobElements = blobElements.OrderBy(blobElement => blobElement.Attribute("Id").Value).ToList();
+
+            var verticalManifest = new XDocument(new XElement(verticalManifestRoot.Name, verticalManifestRoot.Attributes(), packageElements, blobElements));
+
+            File.WriteAllText(VerticalAssetManifestOutputPath, verticalManifest.ToString());
+
+            return true;
```

Comment:
MSBuild should return Log.HasAnyLoggedErrors which is false in case an error is emitted via the Log interface.

```suggestion
            return Log.HasAnyLoggedErrors;
```

Reply:
I could not get this to work. I'm having trouble find any mention of it anywhere too. This task doesn't log anything tho, so it shouldn't make a difference in this case

Reply:
Here's the API doc for it: https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.tasklogginghelper.hasloggederrors?view=msbuild-17-netcore

It should be available...